From 6cc09d5e5f5ddf4d19751345ff40ef463c3821c2 Mon Sep 17 00:00:00 2001
From: Robert Mustacchi <rm@joyent.com>
Date: Mon, 27 Aug 2018 17:15:21 +0000
Subject: [PATCH] OS-7181 cpcgen incorrectly relies on libcmdutils.h Reviewed
 by: Mike Zeller <mike.zeller@joyent.com> Reviewed by: Jason King
 <jason.king@joyent.com> Approved by: Dan McDonald <danmcd@joyent.com>

---
 usr/src/lib/json_nvlist/json_nvlist.h | 2 +-
 usr/src/tools/cpcgen/Makefile         | 2 +-
 usr/src/tools/cpcgen/cpcgen.c         | 3 +++
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/usr/src/lib/json_nvlist/json_nvlist.h b/usr/src/lib/json_nvlist/json_nvlist.h
index ca01ef44f9..75a70f445b 100644
--- a/usr/src/lib/json_nvlist/json_nvlist.h
+++ b/usr/src/lib/json_nvlist/json_nvlist.h
@@ -17,7 +17,7 @@
 #define	_JSON_NVLIST_H
 
 #include <libnvpair.h>
-#include <libcmdutils.h>
+#include <libcustr.h>
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/usr/src/tools/cpcgen/Makefile b/usr/src/tools/cpcgen/Makefile
index 23f412439a..ec38f80e3d 100644
--- a/usr/src/tools/cpcgen/Makefile
+++ b/usr/src/tools/cpcgen/Makefile
@@ -19,7 +19,7 @@ OBJS	= cpcgen.o json_nvlist.o custr.o
 include ../Makefile.tools
 
 LDLIBS	+= -lnvpair
-CPPFLAGS += -I$(SRC)/lib/json_nvlist/
+CPPFLAGS += -I$(SRC)/lib/json_nvlist/ -I$(SRC)/lib/libcustr/common
 
 all: $(PROG)
 
diff --git a/usr/src/tools/cpcgen/cpcgen.c b/usr/src/tools/cpcgen/cpcgen.c
index daa9c255f9..d876b913fc 100644
--- a/usr/src/tools/cpcgen/cpcgen.c
+++ b/usr/src/tools/cpcgen/cpcgen.c
@@ -30,6 +30,9 @@
 #include <sys/param.h>
 #include <assert.h>
 #include <ctype.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 
 #include <json_nvlist.h>
 
-- 
2.21.0

